0.data_analysis_and_processing/nbconverted/filter_nuclei.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Filter Single-Cell Profiles
+# Single-cell profiles are filtered by nuclei bounding box dimensions across all plates.
+# Filtered is accomplished with robust z-score thresholding of nuclei bounding box dimensions.
+
+# In[1]:
+
+
+import pathlib
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+# Remove categorical warning in seaborn
+warnings.filterwarnings("ignore", category=FutureWarning)
+
+
+# # Inputs
+
+# In[3]:
+
+
+data_dir = root_dir / "nuclear_speckles_data"
+nuclear_mask_dir = (data_dir / "Nuclear_masks").resolve(strict=True)
+sc_profiles_path = list((data_dir / "Preprocessed_data/single_cell_profiles").resolve(strict=True).glob("*annotated*.parquet"))
+
+# Load single-cell profile data
+scdfs = [pd.read_parquet(sc_path) for sc_path in sc_profiles_path if sc_path.is_file()]
+
+
+# # Outputs
+
+# In[4]:
+
+
+filtered_sc_path = pathlib.Path("filtered_single_cells")
+filtered_sc_figure_path = filtered_sc_path / "filtered_sc_figures"
+
+filtered_sc_figure_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[5]:
+
+
+def filter_bounding_box_size(_scdf, _bounding_box_col):
+    """
+    Filter nuclei if the robust z score of a bounding box dimension is less than three.
+
+    Parameters
+    ----------
+    _scdf: Pandas Dataframe
+        The unfiltered single-cell data containing bounding box dimensions.
```

Comment:
I would recommend specifying that it is the pycytominer annotated output that this function expects, since the bounding box dimensions are only accurate in that output.

Reply:
This could be computed for other bounding box coordinates as well. Maybe I'll mention pycytominer at the top of the file